core/RankingQuery.php

```diff
@@ -92,11 +92,12 @@ class RankingQuery
     /**
      * Constructor.
      *
-     * @param int|false $limit The result row limit. See {@link setLimit()}.
+     * @param int $limit The result row limit. See {@link setLimit()}.
      */
-    public function __construct($limit = false)
+    public function __construct($limit = 0)
     {
-        if ($limit !== false) {
+        $limit = intval($limit);
```

Comment:
you should always prefer using `(int) $x` over `intval($x)`. There might be no obvious difference, but a `(int)` casting is around 100% faster...